### DISCUSSION TITLE
eslint-plugin-react

### DISCUSSION THREAD

**Discussion by lourot** [vikejs/bati#203] [Q&A] [answered]
When scaffolding with `pnpm create @batijs/app --react --eslint`, the resulting project doesn't include `eslint-plugin-react`, so my JSX doesn't get much validation and I can easily mistype HTML attributes in there, for example.

Should we add it? 

----------

If we decide to go for it, here is what is needed:

1. add `eslint-plugin-react` to `devDependencies`
2. extend `.eslintrc.json` as follows:

```diff
 {
   "root": true,
   "parser": "@typescript-eslint/parser",
-  "extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
+  "extends": [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:react/recommended"
+  ],
   "plugins": ["@typescript-eslint"],
   "parserOptions": {
     "sourceType": "module",
     "ecmaVersion": "latest"
+  },
+  "settings": {
+    "react": {
+      "version": "detect"
+    }
   }
 }
```

(you get a warning when running eslint if you don't add the react version settings, pointing you at https://github.com/jsx-eslint/eslint-plugin-react#configuration )

**Comment by magne4000** [comment]
👍 yes we probably should!

**Comment by magne4000** [comment, marked as answer]
Fixed by https://github.com/batijs/bati/pull/204
